lib/Sema/TypeCheckConstraints.cpp

```diff
@@ -813,6 +815,18 @@ namespace {
         return finish(true, TC.resolveDeclRefExpr(unresolved, DC));
       }
 
+      // For method selectors with unresolved sub-expressions,
+      // resolve the expression but ignore local variables to allow
+      // conflicting method names
+      if (auto selector = dyn_cast<ObjCSelectorExpr>(expr)) {
+        if (selector->isMethodSelector()) {
```

Comment:
Well, correct me if I'm wrong but if it's a getter/setter selector, ignoring local variables doesn't make sense so there is only the method selector left where conflicts can occur with local variable names.